awscrt/io.py

```diff
@@ -96,28 +87,22 @@ def on_shutdown():
 
     @staticmethod
     def get_or_create_static_default():
-        global _static_event_loop_group
-        if '_static_event_loop_group' in globals():
-            if isinstance(_static_event_loop_group, EventLoopGroup):
-                return _static_event_loop_group
-
-        os_cpu_count = os.cpu_count()
-        if os_cpu_count is None:
-            os_cpu_count = 1  # could not find the processor count - default to 1
-        else:
-            # os.cpu_count returns all cores on the system (including virtual ones)
-            # so we divide by two to avoid using all cores on the entire system.
-            os_cpu_count = max(1, int(os_cpu_count / 2))
-
-        _static_event_loop_group = EventLoopGroup(os_cpu_count)
-        return _static_event_loop_group
+        EventLoopGroup._static_event_loop_group_lock.acquire()
+        if EventLoopGroup._static_event_loop_group is not None:
+            if isinstance(EventLoopGroup._static_event_loop_group, EventLoopGroup):
```

Comment:
[style] this insinstance() check isn't necessary. this is a private member, you can assume it's the correct type

in general, python programmers don't check that things are the expected type. They just go ahead and expect things to work, and if they don't work then an exception will bubble up

I know you've seen a lot of isinstance() checks in this codebase, but we only have those to check arguments that the user passes into public functions, and we do that because we pass a lot of things down to C, and the exceptions and stack traces aren't as clear to the user when they come from C, so we make sure the exception happens early